lib/user_repository.rb

```diff
@@ -0,0 +1,61 @@
+require_relative './user'
+
+class UserRepository
+  def all
+    users = []
+
+    sql = "SELECT * FROM users;"
+    result_set = DatabaseConnection.exec_params(sql, [])
+    result_set.each do |record|
+      user = User.new
```

Comment:
The code creating a new `User` object and assigning values from the hash returned by the DB is reused a few times in the other methods — maybe you could factor it into a private method (it could be named `convert_hash_to_user_object`) to simplify / DRY the code?